cgi-bin/ljtestlib.pl

```diff
@@ -0,0 +1,83 @@
+package LJ;
+
+use strict;
+no warnings 'uninitialized';
+
+
+BEGIN {
+    use Test::MockObject;
+
+    my $mock = Test::MockObject->new();
+
+    sub fake_lang_ml {
+	my ( $code, $vars ) = @_;
+	my $aopts = $vars->{'aopts'};
+	if ( $code == "cleanhtml.error.markup.extra" ) {
+	    return "[<strong>Error:</strong> Irreparable invalid markup ('&lt;" . $aopts . "&gt;') in entry. Owner must fix manually. Raw contents below.]";
+	}
+	return "[[[" . $code . "]]]". $aopts;
+	if ( $code == "cleanhtml.error.markup" ) {
+	    return '<a ' . $aopts .'>Error: Irreparable invalid markup in entry. Raw contents behind the cut.</a>';
+	}
+    }
+
+    sub fake_get_proxy_url {
+      return "http://proxy.url";
+    }
+
+    $mock->fake_module('LJ::Lang' => (
+	    ml => \&fake_lang_ml 
+	));
+    $mock->fake_module('DW::Proxy' => (
+	    get_proxy_url => \&fake_get_proxy_url
+	));
+    $mock->fake_module('LJ::EmbedModule' => ());
+
+    # ugly hack to shutup dependent libraries which sometimes want to bring in
+    # ljlib.pl (via require, ick!).  so this lets them know if it's recursive.
+    # we REALLY need to move the rest of this crap to .pm files.
+
+    # ensure we have $LJ::HOME, or complain very vigorously
+    $LJ::HOME ||= $ENV{LJHOME};
+    die "No \$LJ::HOME set, or not a directory!\n"
+        unless $LJ::HOME && -d $LJ::HOME;
+
+    use lib ( $LJ::HOME || $ENV{LJHOME} ) . "/extlib/lib/perl5";
+
+    # Please do not change this to "LJ::Directories"
+    require $LJ::HOME . "/cgi-bin/LJ/Directories.pm";
+}
+
+
+sub is_web_context {
+    return $ENV{MOD_PERL} ? 1 : 0;
```

Comment:
Is this ever possibly true in your anticipated usage of `ljtestlib.pl`?

Reply:
no, you're right, i should just return `0`